wp-content/plugins/wporg-learn/js/course-completion-meta/index.js

```diff
@@ -0,0 +1,56 @@
+/**
+ * WordPress dependencies
+ */
+import { PanelRow, TextControl } from '@wordpress/components';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { PluginDocumentSettingPanel } from '@wordpress/edit-post';
+import { __ } from '@wordpress/i18n';
+import { registerPlugin } from '@wordpress/plugins';
+
+const CourseCompletionMeta = () => {
+	const postMetaData = useSelect( ( select ) => select( 'core/editor' ).getEditedPostAttribute( 'meta' ) || {} );
+	const { editPost } = useDispatch( 'core/editor' );
+
+	const message = postMetaData?._course_completion_success_message || '';
+	const link = postMetaData?._course_completion_survey_link || '';
+
+	return (
+		<PluginDocumentSettingPanel title={ __( 'Course Completion Settings', 'wporg-learn' ) }>
+			<PanelRow>
+				<p>{ __( 'If the fields are left blank, the default values will be applied.', 'wporg-learn' ) }</p>
+			</PanelRow>
+			<PanelRow>
+				<TextControl
+					label={ __( 'Success Message', 'wporg-learn' ) }
+					value={ message }
+					onChange={ ( newMessage ) => {
+						editPost( {
+							meta: {
+								...postMetaData,
+								_course_completion_success_message: newMessage,
+							},
+						} );
+					} }
+				/>
+			</PanelRow>
+			<PanelRow>
+				<TextControl
+					label={ __( 'Survey Link', 'wporg-learn' ) }
```

Comment:
Given this is displayed on the course edit screen in the settings panel I think we can simplify the text a bit, how about
```suggestion
		<PluginDocumentSettingPanel title={ __( 'Completion screen', 'wporg-learn' ) }>
			<PanelRow>
				<p>{ __( 'If the fields are left blank, the default values will be applied.', 'wporg-learn' ) }</p>
			</PanelRow>
			<PanelRow>
				<TextControl
					label={ __( 'Success Message', 'wporg-learn' ) }
					value={ message }
					onChange={ ( newMessage ) => {
						editPost( {
							meta: {
								...postMetaData,
								_course_completion_success_message: newMessage,
							},
						} );
					} }
				/>
			</PanelRow>
			<PanelRow>
				<TextControl
					label={ __( 'Survey Link', 'wporg-learn' ) }
```